scripts/ci-entrypoint.sh

```diff
@@ -251,11 +296,6 @@ cleanup() {
 }
 
 on_exit() {
-    if [[ -n ${KUBECONFIG:-} ]]; then
-        "${KUBECTL}" get nodes -owide || echo "Unable to get nodes"
```

Comment:
I think we're loosing information by removing this? Because the script may fail and be interrupted at any point, before reaching the `wait_for_nodes` and `wait_for_pods` functions getting called, whereas this always run (we're doing `trap on_exit EXIT` below)

Reply:
Yeah, good question. I removed it because I wasn't sure of the value of printing the nodes/pods in the following scenarios that this trapped function cover:

- at the *very* end of a successful test run (like, long after capz has actually built a cluster and perhaps some other things have been done to it)
- at some failure point in the script, perhaps before a cluster has even been built?

Perhaps there is some contract between the ci-entrypoint user and this script that the user expects ci-entrypoint to log a terminal "this is what the cluster looks like at a glance, at the very end of the test run", but it's not clear why a ci-entrypoint-backed test wouldn't do that on its own.

Here's an example of a successful run using the current implementation of vanilla ci-entrypoint.sh run (kicked off via CAPZ PR CI) where there's some "kubectl get nodes stuttering" (it's presented twice).

It does seem like getting rid of this `k get pods -A` statement in the exit func removes that info from test stdout history. If we want that, I'd advocate we put that as a successful logging outcome of the new `wait_for_pods` func.

Basically, as far as ci-entrypoint.sh is concerned, it's only able to provide debug output for its own operations before we hit `on_exit`.

Anyways, those are my thoughts, open to other ideas!

Reply:
maybe the outputs of these should be moved to the logger as files that get saved in _artifacts? in the absence of that, it's definitely helped me debug a few failures "at a glance"

Reply:
ACK, I'll add it back, and add pods output to `wait_for_pods` as well, if folks find this info useful then it can't hurt to have a before/after display of nodes/pods

Reply:
done